src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -978,6 +979,19 @@ CHIP_ERROR ThermostatAttrAccess::AppendPendingPreset(Delegate * delegate, const
     return delegate->AppendToPendingPresetList(preset);
 }
 
+void ThermostatAttrAccess::OnFabricRemoved(const FabricTable & fabricTable, FabricIndex fabricIndex)
+{
+    for (size_t i = 0; i < ArraySize(mAtomicWriteNodeIds); ++i)
+    {
+        auto nodeId = mAtomicWriteNodeIds[i];
+        if (nodeId.GetFabricIndex() == fabricIndex)
+        {
+            mAtomicWriteNodeIds[i] = ScopedNodeId();
+            mAtomicWriteState[i]   = false;
```

Comment:
Followup (because it's not 100% clear what the best way to solve this is): when an endpoint is disabled, we also need to resetAtomicWrite() on that endpoint, right?  Otherwise we can end up in a bad state where the endpoint id gets reused but we still have an atomic state stored for that endpoint it.

Reply:
Added ember callback for shutdown, which calls resetAtomicWrite. Not sure how to write a test for this one.

Reply:
Please make followups be followups so I don't have to keep reviewing the PR over and over again as major changes happen to it.